### DISCUSSION TITLE
Recently started getting "Error: aborted" randomly

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#39242] [Help] [answered]
So I randomly started getting the following error

```shell
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error: aborted
    at connResetException (node:internal/errors:692:14)
    at abortIncoming (node:_http_server:602:17)
    at socketOnClose (node:_http_server:596:3)
    at Socket.emit (node:events:539:35)
    at Socket.emit (node:domain:475:12)
    at TCP.<anonymous> (node:net:709:12)
error - unhandledRejection: Error: aborted
    at connResetException (node:internal/errors:692:14)
    at abortIncoming (node:_http_server:602:17)
    at socketOnClose (node:_http_server:596:3)
    at Socket.emit (node:events:539:35)
    at Socket.emit (node:domain:475:12)
    at TCP.<anonymous> (node:net:709:12) {
  code: 'ECONNRESET'
}
```

It happens if I make my browser go from mobile view to desktop view using Chrome's DevTools and sometimes it will happen just refreshing the page. The only `async` code I have is either `getStaticProps` where I call `sanityClient.fetch(someQuery)` but that has never caused this before. I know this doesn't have any code to go off but that is because I have no idea where to even start.

**Comment by MehmetFaahem** [comment, marked as answer]
just update next version to canary-9 
npm command: npm i next@12.2.4-canary.9

**Reply by JamesSingleton** [reply]
What is in that version that fixes this?

**Reply by MehmetFaahem** [reply]
canary.9

**Reply by JamesSingleton** [reply]
🤦🏼‍♂️ yes I realize there is a `canary.9` but when IN that version fixes it is what I am asking...

**Reply by Executioner1939** [reply]
Yes, knowing what was wrong and what the fix is would help, this is currently causing our Pods to reboot in Production and I can't figure out why. 

As soon as this error pops up K8S kills the Pod.

**Reply by TomSmedley** [reply]
I've also started getting this issue recently. I would also like to know which version fixes this.

**Reply by SugarmanZhu** [reply]
@TomSmedley Updating next to 12.2.4-canary.9 fixes the problem for me.

**Reply by JamesSingleton** [reply]
Yeah updating to that canary version and beyond fixed it… I would just like to know what caused it and what fixed it.

**Reply by AkifB37** [reply]
thx, drove me crazy.

**Reply by niemal** [reply]
It certainly drove me crazy as well, thanks.

**Reply by eric-burel** [reply]
Have this issue with 12.3.0 as well, seems fixed in 12.3.1

**Reply by meyer9** [reply]
Also was getting this in 12.3.0

**Reply by mahmoud-hadi** [reply]
I'm on **12.3.1** and still facing this error randomly.

**Reply by UnholyPhoenix** [reply]
I am also using 12.3.1 and this error happens a few times a day in our production. Should I downgrade to 12.2.4-canary.9?

**Reply by tlenex** [reply]
also facing this isssue on 12.3.3

**Comment by nvh3011** [comment]
also facing the issue on 13.4.1

**Comment by xiaolongkipsi** [comment]
face the same issue in latest version 13.5

**Comment by Neo-Ciber94** [comment]
Same in 13.5.4 using the app router, but only appears to happen when the request is cancelled but do not occur on GET requests

**Reply by Neo-Ciber94** [reply]
Still happening in 14.0.2

**Comment by golddydev** [comment]
I think this is due to the middleware.
I only have this issue, after I added middleware.
I fixed by removing middleware and adding custom logic to api.

**Reply by JamesSingleton** [reply]
I have no middleware in any of my projects though. Granted I haven't run into this issue in some time now/haven't looked for it.

**Comment by marcelino-borges** [comment]
Same here (NextJS 14.0.4)
![image](https://github.com/vercel/next.js/assets/33501037/b893b9df-3aac-49ba-8d1d-4ea188e53c1d)

**Reply by svey-xyz** [reply]
I am having the same issue with 14.1.0

**Comment by brycedbjork** [comment]
I am also experiencing this with 14.1.0

**Reply by jonnitg** [reply]
still happening in 14.2.3 ... it suddenly happened  1 min was ok and the next dev it was failing without any changes

**Comment by sokactomas** [comment]
I am having the same issue with 13.5.6

```sh
AbortError: The operation was aborted.
    at new DOMException (node:internal/per_context/domexception:53:5)
    at Object.destroy (node:internal/deps/undici/undici:10767:33)
    at httpFetch (node:internal/deps/undici/undici:10563:45)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async node:internal/deps/undici/undici:10342:20
    at async mainFetch (node:internal/deps/undici/undici:10332:20)
    ```
It happens only sometimes and only on production.

**Comment by faisal-saddique** [comment]
Following for updates

**Comment by Arhan13** [comment]
Facing the same issue on next 14.1.0

**Comment by work7z** [comment]
experiencing the same issue in next.js v14+

by the way, it's thrown in production env which is deployed as standalone mode

**Comment by Elham13** [comment]
Facing the same issue in next js 14.2.3

![image](https://github.com/vercel/next.js/assets/60072659/2868dfdc-334f-4f1b-bc30-2dbb4f86597f)

**Reply by bugswritter** [reply]
getting the same while using server actions

**Reply by redbmk** [reply]
Were you able to at least trace it down to the calling function? I'm seeing this in 14.2.3 as well and have no clue where it's coming from

**Reply by ignacionar** [reply]
Same for me in v14.2.4

**Comment by GainorB** [comment]
Facing the same issue with `^13.1.6`

**Comment by Imunal** [comment]
Anyone find solution to this problem? Or wait is causing this error?

I'm running local server in Docker image (webServer in playwright config) when running E2E test in CI with Playwright, half of the time tests are failing because server is crashing with "Error: aborted", and needing to be reruned when failing and be prayed to not crash again :)

**Comment by ljhskyso** [comment]
any updates on this? not even sure what triggers the issue...

**Comment by magoz** [comment]
I'm also getting the error. I'm using middleware to setup CORS:

```typescript
// https://nextjs.org/docs/app/building-your-application/routing/middleware#cors

const corsOptions = {
  'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
  'Access-Control-Allow-Headers': 'Content-Type, Authorization'
}

const corsMiddleware = (request: NextRequest) => {
  const origin = '*'
  const isAllowedOrigin = true

  // Handle preflighted requests
  const isPreflight = request.method === 'OPTIONS'

  if (isPreflight) {
    const preflightHeaders = {
      ...(isAllowedOrigin && { 'Access-Control-Allow-Origin': origin }),
      ...corsOptions
    }
    return NextResponse.json({}, { headers: preflightHeaders })
  }

  // Handle simple requests
  const response = NextResponse.next()

  if (isAllowedOrigin) {
    response.headers.set('Access-Control-Allow-Origin', origin)
  }

  Object.entries(corsOptions).forEach(([key, value]) => {
    response.headers.set(key, value)
  })

  return response
}
```

**Comment by Imunal** [comment]
Yeah, anything in middleware will throw this error, but i still didn't find the workaround ;/ Maybe someone have better luck.

Any chance to take a look @leerob ?

**Comment by work7z** [comment]
frankly, my suggestion is not to use next.js   

any leading-edge technology needs time to mitigate these issues

**Reply by teddarcuri** [reply]
This take isn't very constructive, but I actually agree with it.
I have invested my company in Next, but have some regrets. It's like every other day I have a new error cropping up - only to search for issues like this where there is no resolution, nor response.

**Reply by kvnzrch** [reply]
> This take isn't very constructive, but I actually agree with it. I have invested my company in Next, but have some regrets. It's like every other day I have a new error cropping up - only to search for issues like this where there is no resolution, nor response.

This was working fine until Ctrl+S starts the infinite log `POST/200 in 18ms`, and when I reload the page with F5 it throws:
```
⨯ Error: aborted
    at abortIncoming (node:_http_server:797:17)
    at socketOnClose (node:_http_server:791:3)
    at Socket.emit (node:events:531:35)
    at TCP.<anonymous> (node:net:339:12)
digest: "3678400141"
```

I can't get rid of it, I've tried almost everything. It's full of bugs like this. I can't believe this is a "production framework", even recommended by React itself.

**Reply by Thomas-QCD** [reply]
> ```
> ⨯ Error: aborted
>     at abortIncoming (node:_http_server:797:17)
>     at socketOnClose (node:_http_server:791:3)
>     at Socket.emit (node:events:531:35)
>     at TCP.<anonymous> (node:net:339:12)
> digest: "3678400141"
> ```
Currently having this issue on "^14.2.7".

**Comment by trungtin** [comment]
I have this issue, the cause is I'm streaming the response from server to client. After the streaming finish, there some tasks need to be done. So adding `waitUntil`  from '@vercel/functions` to wait for these promises resolve this issue (this apply to local dev environment too)

**Comment by richg0ld** [comment]
15.0.3 version same issue

```bash
[22:52:50.823] ERROR (next.js): aborted
    err: {
      "type": "Error",
      "message": "aborted",
      "stack":
          Error: aborted
              at abortIncoming (node:_http_server:794:17)
              at socketOnClose (node:_http_server:788:3)
              at Socket.emit (node:events:530:35)
              at TCP.<anonymous> (node:net:337:12)
              at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
      "code": "ECONNRESET"
    }
[22:52:50.826] ERROR (next.js):  ⨯ uncaughtException:
    err: {
      "type": "Error",
      "message": "aborted",
      "stack":
          Error: aborted
              at abortIncoming (node:_http_server:794:17)
              at socketOnClose (node:_http_server:788:3)
              at Socket.emit (node:events:530:35)
              at TCP.<anonymous> (node:net:337:12)
              at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
      "code": "ECONNRESET"
    }
[22:52:50.826] ERROR (next.js):  ⨯ uncaughtException:
    err: {
      "type": "Error",
      "message": "aborted",
      "stack":
          Error: aborted
              at abortIncoming (node:_http_server:794:17)
              at socketOnClose (node:_http_server:788:3)
              at Socket.emit (node:events:530:35)
              at TCP.<anonymous> (node:net:337:12)
              at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
      "code": "ECONNRESET"
    }

```

**Comment by gary-jensen** [comment]
> This was working fine until Ctrl+S

Error: aborted
at abortIncoming (node:_http_server:809:17)
at socketOnClose (node:_http_server:803:3)
at Socket.emit (node:events:536:35)
at TCP. (node:net:350:12)
at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
code: 'ECONNRESET'

Ctrl+S recreates this issue for me. No infinite loop here though.
It stems from the two occurences of fetch that I have in my custom hook.
If I either comment out the fetch statement or add an event listener for ctrl+s and preventDefault(), the issue goes away.

I'm not sure what's causing fetch to have this issue though.

**Comment by Hoiy** [comment]
Next 15.1.7: Not sure if this is the same issue. I occationally get one line of log in the local dev env.
```
⨯ [Error: aborted] { code: 'ECONNRESET', digest: '3457017098' }
```
This happens when a request get interrupted at the client side, e.g. by closing the browser. It is not a server issue imo but it appears as an error in the server log.

**Comment by alexandrtovmach** [comment]
Solution:
```diff
if (isPreflight) {
  const preflightHeaders = {
    ...(isAllowedOrigin && { 'Access-Control-Allow-Origin': origin }),
    ...corsOptions
  }
-  return NextResponse.json({}, { headers: preflightHeaders })
+  return NextResponse.next({ headers: preflightHeaders })
}
```

Created PR #76927 to fix it in docs with CORS example as well.

**Comment by SiegeSailor** [comment]
I had this issue earlier today (`"next": "15.3.4"`). I had an API that triggers a `.sh` file to fetch environment variables as a workaround to get variables passed from Docker `--env`. Theoretically, the API should complete the request right away since it runs a local script. However, Next.js seems to send some kind of preflight requests. I constantly saw the error message `⨯ [Error: aborted] { code: 'ECONNRESET', digest: '3678400141' }` every time I updated my code in the development, showing after every page update.
